lib/ansible/modules/cloud/amazon/s3_bucket.py

```diff
@@ -278,6 +285,9 @@ def create_or_update_bucket(s3_client, module, location):
             tags = dict((to_text(k), to_text(v)) for k, v in tags.items())
             if current_tags_dict != tags:
                 if tags:
+                    if not purge_tags:
```

Comment:
Maybe this should be done above `if tags:` on line 287 since `tags: {}` and `purge_tags: False` could be used together.

Reply:
My thinking was that it would be weird to explicitly set `tags: {}` and `purge_tags: False` together - but it could happen when tags is a variable, so thanks for making me reconsider!

Reply:
Handling this and adding some extra tests around it uncovered some additional bugs, so this suggestion helped in more ways than one, thanks!